mitodl/mit-open
# Assign best date available to LearningResourceRun.start_date field

### What are the relevant tickets?
Closes #512 

### Description (What does it do?)
- For OCW courses, assigns a value to `LearningResourceRun.start_date` based on semester/year if present.
- For other ETL sources, if a start date value from source data is null, will substitute  the enrollment start date instead as the value.
- Modifies the old `semester_year_to_date` function a bit in terms of returned values, removing unused `ending` kwarg.


### How can this be tested?
Run `backpopulate_ocw_data --skip-contentfiles`.  It will take ~20-30 minutes but when done nearly all OCW resources should have runs with a populated start date.
Run `backpopulate` commands for other sources too (xpro, mitxonline, oll, etc).  The courses should be imported/updated and most should have runs with start dates.
